File: NewsPortal.Core/Models/News.cs
public class News
{
    public const int MAX_LENGTH = 250;

    private News(Guid id, string title, Guid categoryId, DateTime createdAt, string shortPhrase, string description)
    {
        Id = id;
        Title = title;
        CategoryId = categoryId;
        CreatedAt = createdAt;
        ShortPhrase = shortPhrase;
        Description = description;
    }

    public Guid Id { get; }
    public string Title { get; } = string.Empty;
    public Guid CategoryId { get; }
    public DateTime CreatedAt { get; }
    public string ShortPhrase { get; } = string.Empty;
    public string Description { get; } = string.Empty;

    public static (News news, string Error) Create(
        Guid id,
        string title,
        Guid categoryId,
        DateTime createdAt,
        string shortPhrase,
        string description)
    {
        var error = string.Empty;
        if (string.IsNullOrEmpty(title) || string.IsNullOrEmpty(shortPhrase) || string.IsNullOrEmpty(description))
        {
            error = "Некоторые строки пустые";
        }
        if (title.Length > MAX_LENGTH || shortPhrase.Length > MAX_LENGTH)
        {
            error = "Название и фраза не должны быть больше 250 символов";
        }

        var news = new News(id, title, categoryId, createdAt, shortPhrase, description);
        return (news, error);
    }
}

File: NewsPortal.Frontend/src/app/services/news.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { catchError } from 'rxjs/operators';
import { AuthService } from './auth';

export interface NewsArticle {
  id: string;
  title: string;
  categoryId: string;
  shortPhrase: string;
  description: string;
  createdAt: string;
}

export interface NewsRequest {
  title: string;
  categoryId: string;
  shortPhrase: string;
  description: string;
}

@Injectable({
  providedIn: 'root'
})
export class NewsService {
  private apiUrl = '/api/news';

  constructor(
    private http: HttpClient,
    private authService: AuthService
  ) { }

  private getAuthHeaders(): HttpHeaders {
    const token = this.authService.getToken();
    return new HttpHeaders({
      'Authorization': `Bearer ${token}`,
      'Content-Type': 'application/json'
    });
  }

  getNews(): Observable<NewsArticle[]> {
    return this.http.get<NewsArticle[]>(this.apiUrl).pipe(
      catchError(error => {
        console.error('Ошибка получения новостей:', error);
        throw error;
      })
    );
  }
  createNews(newsRequest: NewsRequest): Observable<string> {
    return this.http.post<string>(this.apiUrl, newsRequest, {
      headers: this.getAuthHeaders()
    }).pipe(
      catchError(error => {
        console.error('Ошибка создания новости:', error);
        throw error;
      })
    );
  }

  updateNews(id: string, newsRequest: NewsRequest): Observable<string> {
    return this.http.put<string>(`${this.apiUrl}/${id}`, newsRequest, {
      headers: this.getAuthHeaders()
    }).pipe(
      catchError(error => {
        console.error('Ошибка обновления новости:', error);
        throw error;
      })
    );
  }

  deleteNews(id: string): Observable<string> {
    return this.http.delete<string>(`${this.apiUrl}/${id}`, {
      headers: this.getAuthHeaders()
    }).pipe(
      catchError(error => {
        console.error('Ошибка удаления новости:', error);
        throw error;
      })
    );
  }
  getNewsById(id: string): Observable<NewsArticle> {
    return this.http.get<NewsArticle>(`${this.apiUrl}/${id}`).pipe(
      catchError(error => {
        console.error('Ошибка получения новости:', error);
        throw error;
      })
    );
  }
}
